src/python/pants/backend/go/util_rules/build_pkg_target.py

```diff
@@ -579,6 +580,7 @@ async def resolve_go_stdlib_embed_config(
             input_digest=input_digest,
             description=f"Create embed mapping for {request.package.import_path}",
             level=LogLevel.DEBUG,
+            immutable_input_digests={".goroot": goroot.digest},
```

Comment:
Maybe have a `goroot.immutable_input_digests` function which returns this dictionary?